server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/StringRareTermsAggregator.java

```diff
@@ -63,96 +87,106 @@ public LeafBucketCollector getLeafCollector(LeafReaderContext ctx,
             final BytesRefBuilder previous = new BytesRefBuilder();
 
             @Override
-            public void collect(int docId, long bucket) throws IOException {
-                assert bucket == 0;
-                if (values.advanceExact(docId)) {
-                    final int valuesCount = values.docValueCount();
-                    previous.clear();
-
-                    // SortedBinaryDocValues don't guarantee uniqueness so we
-                    // need to take care of dups
-                    for (int i = 0; i < valuesCount; ++i) {
-                        final BytesRef bytes = values.nextValue();
-                        if (includeExclude != null && !includeExclude.accept(bytes)) {
-                            continue;
-                        }
-                        if (i > 0 && previous.get().equals(bytes)) {
-                            continue;
-                        }
-
-                        doCollect(sub, bytes, docId);
-                        previous.copyBytes(bytes);
+            public void collect(int docId, long owningBucketOrd) throws IOException {
+                if (false == values.advanceExact(docId)) {
+                    return;
+                }
+                int valuesCount = values.docValueCount();
+                previous.clear();
+
+                // SortedBinaryDocValues don't guarantee uniqueness so we
+                // need to take care of dups
+                for (int i = 0; i < valuesCount; ++i) {
+                    BytesRef bytes = values.nextValue();
+                    if (filter != null && !filter.accept(bytes)) {
```

Comment:
Nit: `!filter.accept()` :)

(Also I realize the irony since the original code had that and it was my fault :) )

Reply:
:+1: